core/src/test/scala/unit/kafka/tools/DumpLogSegmentsTest.scala

```diff
@@ -831,6 +832,86 @@ class DumpLogSegmentsTest {
     )
   }
 
+  @Test
+  def testTransactionLogMessageParser(): Unit = {
+    val serde = new TransactionCoordinatorRecordSerde()
+    val parser = new TransactionLogMessageParser()
+
+    def serializedRecord(key: ApiMessageAndVersion, value: ApiMessageAndVersion): Record = {
+      val record = new CoordinatorRecord(key, value)
+      TestUtils.singletonRecords(
+        key = serde.serializeKey(record),
+        value = serde.serializeValue(record)
+      ).records.iterator.next
+    }
+
+    // The key is mandatory.
+    assertEquals(
+      "Failed to decode message at offset 0 using offset transaction-log decoder (message had a missing key)",
+      assertThrows(
+        classOf[RuntimeException],
+        () => parser.parse(TestUtils.singletonRecords(key = null, value = null).records.iterator.next)
+      ).getMessage
+    )
+
+    // A valid key and value should work.
```

Comment:
I wonder whether we ought to have some testing of a more complicated value too. There are tagged fields and so on to consider.

Reply:
I've added a testcase with all fields set for `TransactionValue`. Is this what you had in mind?

Reply:
I think that will do. The point is to catch a situation where we inadvertently break this.